### PR TITLE
Rename nudgeunix to drnudgeunix

### PR DESCRIPTION
**Pull request by fmoessbauer**
This patch renames the nudgeunix tool to drnudgeunix to be
named more consistently with the rest of DynamoRIOs frontend tooling.
This helps for packaging as nudgeunix is somehow non-specific.

For backward compatibility, a symlink with the old name is added.

Related to issue #5153

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

### CODE REVIEW

**Comment by derekbruening** [comment]
I don't remember why this is, but drrun has a -nudge option for Windows -- yet for Linux nudging is in a separate program.  Should we remove this separate drnudgeunix and put its functionality into drrun to match Windows?

Happy to rename for now if we want to file an issue on that for later.

File: tools/drdeploy.c [hunk 1]
```diff
@@ -265,8 +265,8 @@ const char *options_list_str =
     "                          wait.  A value of 0 means don't wait for nudges to\n"
     "                          complete."
 #    else  /* WINDOWS */
-    /* FIXME i#840: integrate nudgeunix into drconfig on Unix */
-    "Note: please use the nudgeunix tool to nudge processes on Unix.\n";
+    /* FIXME i#840: integrate drnudgeunix into drconfig on Unix */
```
Comment (derekbruening) [inline tools/drdeploy.c:268]: Ah here's an already-filed issue on my top-level comment: ok let's leave it for that issue.
No action requested.  (Githhub should add a type of comment that is already resolved.)

File: tools/drdeploy.c [hunk 2]
```diff
@@ -265,8 +265,8 @@ const char *options_list_str =
...
+    "Note: please use the drnudgeunix tool to nudge processes on Unix.\n";
```
Comment (derekbruening) [inline tools/drdeploy.c:269]: I'm seeing a bunch of other references that need updating, right?
```
$ git grep nudgeunix
api/docs/deployment.dox:nudgeunix tool:
api/docs/deployment.dox:bin32/nudgeunix -pid targetpid -client 0 5
core/lib/dr_events.h: * nudgeunix tool on Linux.  A client in this process can use
core/unix/nudgesig.c: * shared between core & tools/nudgeunix.c
core/unix/nudgesig.c:/* shared with tools/nudgeunix.c */
suite/tests/CMakeLists.txt:        get_target_path_for_execution(toolbindir nudgeunix "${location_suffix}")
suite/tests/runall.cmake:# * nudge = arguments to nudgeunix or drconfig
suite/tests/runall.cmake:  set(nudge_cmd nudgeunix)
suite/tests/runall.cmake:  # nudgeunix and drconfig have different syntax:
tools/CMakeLists.txt:    add_executable(nudgeunix nudgeunix.c ${PROJECT_SOURCE_DIR}/core/unix/nudgesig.c)
tools/CMakeLists.txt:    DR_install(TARGETS nudgeunix DESTINATION ${INSTALL_BIN})
tools/drdeploy.c:    /* FIXME i#840: integrate nudgeunix into drconfig on Unix */
tools/drdeploy.c:    "Note: please use the nudgeunix tool to nudge processes on Unix.\n";
tools/nudgeunix.c: * nudgeunix.c: nudges a target Linux process
tools/nudgeunix.c:    "usage: nudgeunix [-help] [-v] [-pid <pid>] [-type <type>] [-client <ID> <arg>]\n"
tools/nudgeunix.c:            printf("nudgeunix version %s -- build %d\n", STRINGIFY(VERSION_NUMBER),
```

Comment (fmoessbauer) [reply tools/drdeploy.c:269]: This grep is run on a different source tree. But you are right. I still missed a couple of references.

**Comment by fmoessbauer** [comment]
@derekbruening I just renamed all references, except for the filename (`nudgeunix.c`). Renaming this as well adds only little value but destroys the git history.

**Review by derekbruening — APPROVED** [review]

**Pull request merged**